CRM/Contribute/BAO/Contribution.php

```diff
@@ -3729,4 +3763,31 @@ public static function checkFinancialTypeChange($financialTypeId, $contributionI
     }
   }
 
+  /**
+   * Generate credit note id with next avaible number 
+   *
+   * @return string
+   *   Credit Note Id.
+   */
+  public static function createCreditNoteId() {
+    $prefixValue = CRM_Core_BAO_Setting::getItem(CRM_Core_BAO_Setting::CONTRIBUTE_PREFERENCES_NAME, 'contribution_invoice_settings');
+
```

Comment:
I think next 4 lines can be condensed to:
$creditNoteNum = CRM_Core_DAO::singleValueQuery("SELECT count(creditnote_id) as creditnote_number FROM civicrm_contribution");